student_survey.php

```diff
@@ -0,0 +1,255 @@
+<?php
+require 'survey_query_generators.php';
+header("Content-type: application/json");
+try {
+    $method = "";
+    if (isset($_SERVER['REQUEST_METHOD'])) {
+        $method = $_SERVER['REQUEST_METHOD'];
+        switch ($method) {
+            case "GET":
+                print json_encode(handle_get($_GET));
+                exit();
+            case "POST":
+                print json_encode(handle_post($_POST));
+                exit();
+            default:
+                throw new Exception("Invalid Request");
+        }
+    } else {
+        $error = 'No Request Method Found';
+        throw new Exception($error);
+    }
+} catch (Exception $e) {
+    $result = set_http_response(400);
+    date_default_timezone_set('America/Toronto');
+    error_log(
+        date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
+        3,
+        "errors.log"
+    );
+    print json_encode($result, JSON_PRETTY_PRINT);
+    exit();
+}
+/**
+ * Function executes sql query and logs any sql error.
+ *
+ * @param query_string sql query string
+ * @param bind_variables an associative array of variables to bind to sql query
+ *
+ */
+function get_query_result($query_string, $bind_variables)
+{
+    require '../db/config.php';
+    // Attempt to execute sql command and print response in json format.
+    // If a sql error occurs, JSON error object.
+    try {
+        // connect to the mysql database
+        $conn = new PDO(
+            "mysql:host=$servername;dbname=$database",
+            $username,
+            $password
+        );
+        // set the PDO error mode to exception
+        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        $stmt = $conn->prepare($query_string);
+
+        foreach ($bind_variables as $key => $value) {
+            $stmt->bindValue($key, $value);
+        }
+        $stmt->execute();
+        //fetch all results
+        if ($_SERVER['REQUEST_METHOD'] == 'GET') {
+            $fetched = $stmt->fetchAll(PDO::FETCH_ASSOC);
+            return $fetched;
+        } else {
+            return array("TYPE" => "Success");
+        }
+    } catch (PDOException $e) {
+        $result = set_http_response(500);
+        date_default_timezone_set('America/Toronto');
+        error_log(
+            date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
```

Comment:
You need `\n` in the error log?

Reply:
Yes or else every error message would display on the same line in errors.log. It's a carry over from get_info.php.